### PR TITLE
docs: reformat config reference

### PR DESCRIPTION
**Pull request by TimJones**
Update the configuration reference documentation
to show field information in a tabular format.

Signed-off-by: Tim Jones <tim.jones@siderolabs.com>

Example of the tabular layout:
image

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5440)

### CODE REVIEW

**Comment by smira** [comment]
I don't know whether it was intentional or not, but inline examples are now missing, e.g. for `MachineConfig` `sysctls` field (and all other fields). Without these examples it's hard to see what to put as the values of these fields as they don't have any specific sub-section.

**Comment by smira** [comment]
I wonder if we should follow table with some examples? or have per-field example below the table? I don't know what is the best, just throwing out ideas.

**Comment by TimJones** [comment]
Now with expandable, syntax-highlighted examples!!
Screenshot_20220426_163229

File: pkg/machinery/config/encoder/markdown.go [hunk 1]
```diff
@@ -195,9 +158,18 @@ func encodeYaml(in interface{}, name string) string {
 		lines[i] = strings.TrimRight(line, " ")
 	}
 
-	return fmt.Sprintf("``` yaml\n%s```", strings.Join(lines, "\n"))
+	return fmt.Sprintf("{{< highlight yaml >}}\n%s{{< /highlight >}}", strings.Join(lines, "\n"))
```
Comment (TimJones) [inline pkg/machinery/config/encoder/markdown.go:161]: A possible issue here is that this is coupling the outputted markdown file with Hugo's rendering engine.

**Comment by frezbo** [comment]
> Now with expandable, syntax-highlighted examples!! Screenshot_20220426_163229

that is some vodoo magic :mage:

**Review by smira — APPROVED** [review]
:heart:

**Pull request merged**